proposals/0016-c-interface-compiler-library.md

```diff
@@ -50,7 +50,495 @@ older DXC-based solutions to adopt clang as the preferred HLSL compiler.
 
 ## Detailed design
 
-_The detailed design is not required until the feature is under review._
+### What is provided today in the DXC api?
+The current DirectX shader compiler library is a nano-COM implementation that
+supports the following features.
+
+* [Include Handlers](#include-handlers)
+* [Compiler](#shader-compiler)
+* [Linker](#shader-linker)
+* [Validation](#shader-validation)
+* [Reflection Data Access](#shader-reflection)
+* [DXIL Container Access](#dxil-container-access)
+* [PDB Symbol Access](#shader-pdbs)
+
+These features will also need to be supported by the new c-style library.
+Code snippets on how to compile a shader using the COM library are located
+in [DXC api code snippets](#dxc-code-snippets).
+
+
+The following interfaces are used to work with data being passed to/from the
+library.
+
+#### Buffers
+```c++
+// Structure for supplying bytes or text input to Dxc APIs. Represents both
+// text (DxcText) and non-text byte buffers (DxcBuffer).
+struct DxcBuffer {
+  LPCVOID Ptr;
+  SIZE_T Size;
+  UINT Encoding;
```

Comment:
Is there a list of valid values for `Encoding` anywhere?  Presumably this comes from a enum or something?

Reply:
Added in latest push